lightning/src/routing/network_graph.rs

```diff
@@ -50,12 +50,63 @@ const MAX_EXCESS_BYTES_FOR_RELAY: usize = 1024;
 /// This value ensures a reply fits within the 65k payload limit and is consistent with other implementations.
 const MAX_SCIDS_PER_REPLY: usize = 8000;
 
+/// Represents the compressed public key of a node
+#[derive(Clone, Copy)]
+pub struct NodeId(pub [u8; 33]);
+
+impl fmt::Debug for NodeId {
+	fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+		write!(f, "NodeId({})", log_bytes!(self.0))
+	}
+}
+
+impl core::hash::Hash for NodeId {
+	fn hash<H: core::hash::Hasher>(&self, hasher: &mut H) {
+		self.0.hash(hasher);
+	}
+}
+
+impl Eq for NodeId {}
+
+impl PartialEq for NodeId {
+	fn eq(&self, other: &Self) -> bool {
+		self.0[..] == other.0[..]
+	}
+}
+
+impl cmp::PartialOrd for NodeId {
+	fn partial_cmp(&self, other: &Self) -> Option<cmp::Ordering> {
+		Some(self.cmp(other))
+	}
+}
+
+impl Ord for NodeId {
+	fn cmp(&self, other: &Self) -> cmp::Ordering {
+		self.0[..].cmp(&other.0[..])
+	}
+}
+
+impl Writeable for NodeId {
+	fn write<W: Writer>(&self, writer: &mut W) -> Result<(), io::Error> {
+		writer.write_all(&self.0)?;
+		Ok(())
+	}
+}
+
+impl Readable for NodeId {
+	fn read<R: io::Read>(reader: &mut R) -> Result<Self, DecodeError> {
+		let mut buf = [0; 33];
+		reader.read_exact(&mut buf)?;
```

Comment:
Should we attempt to parse this as `PublicKey` to ensure it can be later converted to one?